addons/mail/static/src/models/message_reaction_group/message_reaction_group.js

```diff
@@ -0,0 +1,137 @@
+/** @odoo-module **/
+
+import { registerNewModel } from '@mail/model/model_core';
+import { attr, many2many, many2one } from '@mail/model/model_field';
+import { insertAndReplace } from '@mail/model/model_field_command';
+const { markEventHandled } = require('@mail/utils/utils');
+
+function factory(dependencies) {
+
+    class MessageReactionGroup extends dependencies['mail.model'] {
+
+        /**
+         * @override
+         */
+        _created() {
+            super._created();
+            this.onClick = this.onClick.bind(this);
+        }
+
+        //----------------------------------------------------------------------
+        // Public
+        //----------------------------------------------------------------------
+
+        /**
+         * Handles click on the reaction group.
+         *
+         * @param {ev}
+         */
+        onClick(ev) {
+            markEventHandled(ev, 'MessageReactionGroup.Click');
+            if (this.hasUserReacted) {
+                this.message.removeReaction(this.content);
+            } else {
+                this.message.addReaction(this.content);
+            }
+        }
+
+        //----------------------------------------------------------------------
+        // Private
+        //----------------------------------------------------------------------
+
+        /**
+         * @override
+         */
+        static _createRecordLocalId(data) {
+            return `${this.modelName}_${data.messageId}_${data.content}`;
```

Comment:
Are we doing any string processing on local ids? Just want to make sure we don't introduce any unexpected behaviour by adding weird characters in them

Reply:
> Are we doing any string processing on local ids? Just want to make sure we don't introduce any unexpected behaviour by adding weird characters in them (eg split and regex matching can behave weirdly with unicode characters that span mutiple code points)

Conceptually, values for local id shouldn't matter as long as there is no collision. So ideally this override put a more humanly understandable value to help debugging.

In practice, the `findFromIdentifyingData` method relies on `_createRecordLocalId` to get record in O(1), so it's mandatory to override `_createRecordLocalId` correctly or you cannot find record of this model uniquely from identifying data. It's important not to break this, because `insert()` needs that in order to properly update existing records.

Reply:
As explained, no processing done, this could be a meaningless hash but we prefer string for easier debugging. As long as it is unique depending on the data that are considered together as making a unique key.